app/parse.py

```diff
@@ -15,8 +32,293 @@ class Product:
+    driver = webdriver.Chrome()
+    driver.get(TABLETS_URL)
+    cookies = driver.find_element(By.CLASS_NAME, "acceptCookies")
+    if cookies:
+        cookies.click()
+    driver.execute_script("window.scrollTo(0, 1000);")
+    time.sleep(1)
+    more_button = driver.find_element(
+        By.CLASS_NAME,
+        "ecomerce-items-scroll-more"
+    )
+    while more_button:
+        try:
+            WebDriverWait(
+                driver,
+                1
+            ).until(
+                ec.element_to_be_clickable(
+                    (
+                        By.CLASS_NAME,
+                        "ecomerce-items-scroll-more")
+                )
+            )
+            more_button.click()
+            time.sleep(1)
+        except Exception as e:
+            logging.info(f"An error occurred during tablet scraping {e}")
+            break
+    html = driver.page_source
+    soup = BeautifulSoup(html, "html.parser")
+    tablet_tag = soup.select(
+        "div.container > div.row > div.col-lg-9 > div.row > div.col-md-4"
+    )
+
+    return [get_single_tablet(tablet) for tablet in tablet_tag]
+
+
+def get_all_touch() -> list[Product]:
+    driver = webdriver.Chrome()
+    driver.get(TOUCH_URL)
+    cookies = driver.find_element(By.CLASS_NAME, "acceptCookies")
+    if cookies:
+        cookies.click()
+    driver.execute_script("window.scrollTo(0, 1000);")
+    time.sleep(1)
+    more_button = driver.find_element(
+        By.CLASS_NAME,
+        "ecomerce-items-scroll-more"
+    )
+    while more_button:
+        try:
+            WebDriverWait(
+                driver,
+                1
+            ).until(
+                ec.element_to_be_clickable(
+                    (
+                        By.CLASS_NAME,
+                        "ecomerce-items-scroll-more")
+                )
+            )
+            more_button.click()
+            time.sleep(1)
+        except Exception as e:
+            logging.info(f"An error occurred during touch scraping {e}")
+            break
+    html = driver.page_source
+    soup = BeautifulSoup(html, "html.parser")
+    touch_tag = soup.select(
+        "div.container > div.row > div.col-lg-9 > div.row > div.col-md-4"
+    )
+
+    return [get_single_tablet(touch) for touch in touch_tag]
+
+
+async def get_all_home_products(client: AsyncClient) -> list[Product]:
+    page = await client.get(HOME_URL)
+    first_page = BeautifulSoup(page.content, "html.parser")
+    return await get_single_page_home_product(first_page)
+
+
+async def get_all_computers(client: AsyncClient) -> list[Product]:
+    page = await client.get(COMPUTER_URL)
+    first_page = BeautifulSoup(page.content, "html.parser")
+    return await get_single_computer_page(first_page)
+
+
+async def get_all_phones(client: AsyncClient) -> list[Product]:
+    page = await client.get(PHONE_URL)
+    first_page = BeautifulSoup(page.content, "html.parser")
+    return await get_single_phone_page(first_page)
+
```

Comment:
Why not to create one function and pass an url there?

Reply:
If the structure of products can change independently of each other, it’s better to keep separate functions. However, if you’re confident that the parsing logic will always be similar, you can use a generalized function to avoid duplication.

**Optimal solution:** Find a compromise — create a base function for common tasks (like loading pages and parsing) and keep separate functions for specific product processing.

On my mind I'll keep one utility function here